soccer/gameplay/plays/restarts/our_free_kick.py

```diff
@@ -21,25 +30,44 @@ def __init__(self):
         # kicker.use_chipper = True
         kicker.min_chip_range = 0.3
         kicker.max_chip_range = 3.0
+        # This will be reset to something else if indirect on the first iteration
         kicker.target = constants.Field.TheirGoalSegment
         self.add_subbehavior(kicker, 'kicker', required=False, priority=5)
 
         # add two 'centers' that just move to fixed points
         center1 = skills.move.Move(robocup.Point(0, 1.5))
         self.add_subbehavior(center1, 'center1', required=False, priority=4)
         center2 = skills.move.Move(robocup.Point(0, 1.5))
-        self.add_subbehavior(center1, 'center2', required=False, priority=3)
+        self.add_subbehavior(center2, 'center2', required=False, priority=3)
+
+        lastKicker = kicker
+        if self.indirect:
+            receive_pt, target_point, probability = OurFreeKick.tpass.eval_best_receive_point(main.ball().pos)
+            pass_behavior = tactics.coordinated_pass.CoordinatedPass(receive_pt, None, (kicker, lambda x: True))
+            # We don't need to manage this anymore
+            self.remove_subbehavior('kicker')
+
+            self.add_subbehavior(pass_behavior, 'receiver', required=False, priority=5)
+            kicker.target = receive_pt
+            lastKicker = pass_behavior
 
         self.add_transition(
             behavior.Behavior.State.running, behavior.Behavior.State.completed,
-            lambda: kicker.is_done_running(), 'kicker completes')
+            lambda: lastKicker.is_done_running(), 'kicker completes')
+
 
     @classmethod
     def score(cls):
         gs = main.game_state()
-        return 0 if gs.is_ready_state() and gs.is_our_free_kick() else float(
+        return 0 if OurFreeKick.running or (gs.is_ready_state() and gs.is_our_free_kick()) else float(
             "inf")
 
+    def on_enter_running(self):
+        OurFreeKick.running = True
```

Comment:
I was having issues with this play cutting off (so the receiver wouldnt get a chance to actually catch the ball), so I'm writing to a variable here which will force the score to stay at 0 (as long as an instance of this play is running). There is a race condition here, but we won't ever want to run more than one restart at a time, so I think its fine for now (?)

@justbuchanan